src/Dialog/Dialog.tsx

```diff
@@ -310,61 +298,211 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
   const body = (renderBody ?? DefaultBody)(defaultedProps)
   const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
 
-  return (
-    <>
+  if (responsiveType === 'full-screen') {
```

Comment:
There is a lot of duplication going on here. It might benefit from hoisting the element into a variable. Something like:


```js
let TheDialog = NormalDialog;
if (responsiveType === 'full-screen') TheDialog = FullScreenDialog
else if (responsiveType === 'bottom-sheet') TheDialog = BottomSheetDialog
```

I haven't looked at the duplication within each of these components but it might be worth looking into that too.

Reply:
It doesn't have to, but there are subtle differences that cause them to behave differently. When animation is used in the code, it can be easier to understand if you organize the different states by their names. Of course, this is a matter of personal preference, and you could also consider adding an 'isFullScreen' boolean property here.

``` JavaScript
const NormalDialog = styled.div<StyledDialogProps>`
  ...
  box-shadow: ${get('shadows.overlay.shadow')};
  min-width: 296px;
  max-width: calc(100vw - 64px);
  max-height: calc(100vh - 64px);
  ...
  width: ${props => widthMap[props.width ?? ('xlarge' as const)]};
  height: ${props => heightMap[props.height ?? ('auto' as const)]};
  border-radius: 12px;
  ...
  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
      transform: scale(0.5);
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  ${sx};
`

const FullScreenDialog = styled.div<StyledDialogProps>`
  ...
  width: 100vw;
  height: 100vh;
  ...
  animation: overlay--dialog-appear ${ANIMATION_DURATION} ${get('animation.easeOutCubic')};
  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  ${sx};
`
```